core/math/math_2d.h

```diff
@@ -239,22 +239,31 @@ struct Rect2 {
 
 	inline real_t distance_to(const Vector2 &p_point) const {
 
-		real_t dist = 1e20;
+		real_t dist;
+		bool inside = true;
 
 		if (p_point.x < position.x) {
-			dist = MIN(dist, position.x - p_point.x);
+			real_t d = position.x - p_point.x;
+			dist = inside ? d : MIN(dist, d);
```

Comment:
`dist` is used here but it's not initialized, it gives a warning in the compiler.

Reply:
`inside` is always `true` here, so `dist` is never read, still should have initialized `dist` to `0` to get rid of the compiler warning.